lib/Chunk.js

```diff
@@ -121,124 +117,136 @@ class Chunk {
 	}
 
 	remove(reason) {
-		this.modules.slice().forEach(m => {
-			m.removeChunk(this);
-		}, this);
-		this.parents.forEach(c => {
-			const idx = c.chunks.indexOf(this);
-			if(idx >= 0) {
-				c.chunks.splice(idx, 1);
-			}
-			this.chunks.forEach(cc => {
-				cc.addParent(c);
-			});
-		}, this);
-		this.chunks.forEach(c => {
-			const idx = c.parents.indexOf(this);
+		// cleanup modules
+		this.modules.slice().forEach(module => {
+			module.removeChunk(this);
+		});
+
+		// cleanup parents
+		this.parents.forEach(parentChunk => {
+			// remove this chunk from its parents
+			const idx = parentChunk.chunks.indexOf(this);
 			if(idx >= 0) {
-				c.parents.splice(idx, 1);
+				parentChunk.chunks.splice(idx, 1);
 			}
-			this.parents.forEach(cc => {
-				cc.addChunk(c);
+
+			// cleanup "sub chunks"
+			this.chunks.forEach(chunk => {
+				/**
+				 * remove this chunk as "intermediary" and connect
+				 * it "sub chunks" and parents directly
+				 */
+				// add parent to each "sub chunk"
+				chunk.addParent(parentChunk);
+				// add "sub chunk" to parent
+				parentChunk.addChunk(chunk);
```

Comment:
true, i did not think of the first point, can that happen though?

i doubt the old one is more performant, as it iterates over all chunks and then over all parent
and then again over all parents and over all chunks.

this one iterates once over all chunks and all parents.